namespace.go

```diff
@@ -0,0 +1,39 @@
+package vector
+
+import "errors"
+
+const deleteNamespacePath = "/delete-namespace"
+const listNamespacesPath = "/list-namespaces"
+
+// Namespace returns a new Index client for a specific namespace.
+func (ix *Index) Namespace(ns string) (i *Index) {
+	return &Index{
+		url:       ix.url,
+		token:     ix.token,
+		client:    ix.client,
+		namespace: ns,
+		headers:   ix.headers,
+	}
+}
+
+// DeleteNamespace deletes the given namespace of index if it exists.
+func (ix *Index) DeleteNamespace(ns string) error {
+	if ns == "" {
+		return errors.New("cannot delete the default namespace")
+	}
+	pns := ix.namespace
+	ix.namespace = ns
+	_, err := ix.sendBytes(deleteNamespacePath, nil, true)
+	ix.namespace = pns
```

Comment:
I am not sure we ever said that our Index is thread safe, but if we want it to be, this might not work.

Reply:
This logic is removed ✅